src/can.cpp

```diff
@@ -0,0 +1,91 @@
+#include "cand/can.hpp"
+
+CAN::CAN(std::string name):interface_name{name} {
+
+    int s; 
+    struct sockaddr_can addr;
+    struct ifreq ifr;
+
+    if ((s = socket(PF_CAN, SOCK_RAW, CAN_RAW)) < 0) {
+        perror("Socket initialization error");
+        return;
+    }
+
+    interface_name.copy(ifr.ifr_name, interface_name.length(), 0);
+    ioctl(s, SIOCGIFINDEX, &ifr);
+
+    memset(&addr, 0, sizeof(addr));
+    addr.can_family = AF_CAN;
+    addr.can_ifindex = ifr.ifr_ifindex;
+
+    if (bind(s, (struct sockaddr *)&addr, sizeof(addr)) < 0) {
+        perror("Binding error");
+        return;
+    }
+
+    file_descriptor = s;
+
+}
+        
+int CAN::socket_write (struct can_frame *frame) {
+
+    try{
+        int vl = write(file_descriptor, frame, sizeof(struct can_frame));
+        if(vl<0){
+            throw vl;
+        }
+    }catch(int ex){
+        std::cerr<<"Write error";
+        return 1;
+    }
```

Comment:
Em vez desse try catch seria mais legal melhorar esse throw pra já embutir a mensagem de erro
```suggestion
      int vl = write(file_descriptor, frame, sizeof(struct can_frame));
      if(vl<0){
          throw std::runtime_error("Write error");
      }
```

Reply:
[Aqui](https://en.cppreference.com/w/cpp/error) da pra ver mais um pouco sobre como fazer error handling em C++ se achar interessante